### PR TITLE
Add support for multiple fetches at a time

### PR DESCRIPTION
**Pull request by achlee**

### CODE REVIEW

File: src/GatherContent/HTTPClient.php [hunk 1]
```diff
@@ -60,7 +69,80 @@ private function sendRequest($url, $additionalCurlOptions)
...
+
+        // execute - if there is an active connection then keep looping
+        $active = null;
+        do {
+            print('.');
+            $status = curl_multi_exec($mh, $active);
+        }
+        while ($active && $status == CURLM_OK);
+
+//        do {
```
Comment (connorjclark) [inline src/GatherContent/HTTPClient.php]: remove dead code

File: src/GatherContent/HTTPClient.php [hunk 2]
```diff
@@ -60,7 +69,80 @@ private function sendRequest($url, $additionalCurlOptions)
         $response->status = $status;
         $response->body   = $body;
 
+
+        $end = microtime(true);
+
+        print($url.': '.($end - $start).' seconds'.PHP_EOL);
```
Comment (connorjclark) [inline src/GatherContent/HTTPClient.php]: libraries shouldn't print - it makes it impossible for consumers to control the output.

Do we have some sort of common Logging functionality available for our PHP projects?

Comment (achlee) [reply src/GatherContent/HTTPClient.php]: I removed this. Was accidentally left in. We have some logging and if this timing information is useful we can do it higher up the stack. I agree it shouldn't be in the library itself.

File: src/GatherContent/HTTPClient.php [hunk 3]
```diff
@@ -63,4 +73,65 @@ private function sendRequest($url, $additionalCurlOptions)
...
+
+        $active = 0;
+        $urlIndex = 0;
+
+        // Process Loop
+        do {
+            // Add to the in flight requests
+            $numToAdd = $this->totalConcurrentRequests - $active;
+
+            while ($numToAdd-- && $urlIndex < count($urls)) {
```
Comment (connorjclark) [inline src/GatherContent/HTTPClient.php:95]: I think this do...while construct, in addition to "$numToAdd--" in the inner while conditional, could cause an infinite loops if the functions is called with $urls set to an empty array.

Comment (achlee) [reply src/GatherContent/HTTPClient.php:95]: If $urls is empty, $urlIndex < count($urls) is false so it should break out of this on both this loop and the outer do...while loop, right?

Comment (connorjclark) [reply src/GatherContent/HTTPClient.php:95]: and what happens on line 110?

**Comment by pkw** [comment]
OK, looks good.

**Pull request merged**